x/ecocredit/client/tx.go

```diff
@@ -79,59 +82,82 @@ Parameters:
 	}
 }
 
+const (
+	FlagIssuer          string = "issuer"
+	FlagClassId         string = "class-id"
+	FlagStartDate       string = "start-date"
+	FlagEndDate         string = "end-date"
+	FlagProjectLocation string = "project-location"
+	FlagMetadata        string = "metadata"
+	FlagIssuance        string = "issuance"
+)
+
 func txCreateBatch() *cobra.Command {
-	return &cobra.Command{
-		Use:   "create-batch [issuer] [class_id] [start_date] [end_date] [metadata] [issuance]",
+	var (
+		startDate = time.Unix(10000, 10000).UTC()
+		endDate   = time.Unix(10000, 10050).UTC()
+	)
+	createBatchJSON, err := json.MarshalIndent(
+		ecocredit.MsgCreateBatchRequest{
+			// Leave issuer empty, because we'll use --from flag
+			Issuer:  "",
+			ClassId: "1BX53GF",
+			Issuance: []*ecocredit.MsgCreateBatchRequest_BatchIssuance{
+				&ecocredit.MsgCreateBatchRequest_BatchIssuance{
+					Recipient:          "regen1elq7ys34gpkj3jyvqee0h6yk4h9wsfxmgqelsw",
+					TradableAmount:     "1000",
+					RetiredAmount:      "15",
+					RetirementLocation: "ST-UVW XY Z12",
+				},
+			},
+			Metadata:        []byte{0x1, 0x2},
+			StartDate:       &startDate,
+			EndDate:         &endDate,
+			ProjectLocation: "AB-CDE FG1 345",
+		},
+		"                    ",
+		"    ",
+	)
+	if err != nil {
+		panic("Couldn't marshal MsgCreateBatch to JSON")
+	}
+	cmd := &cobra.Command{
+		Use:   "create-batch msg-create-batch",
 		Short: "Issues a new credit batch",
-		Long: `Issues a new credit batch.
+		Long: fmt.Sprintf(`Issues a new credit batch.
 
 Parameters:
-  issuer:     issuer address
-  class_id:   credit class
-  start_date: The beginning of the period during which this credit batch was
-              quantified and verified. Format: yyyy-mm-dd.
-  end_date:   The end of the period during which this credit batch was
-              quantified and verified. Format: yyyy-mm-dd.
-  metadata:   base64 encoded issuance metadata
-  issuance:   YAML encode issuance list. Note: numerical values must be written in strings.
-              eg: '[{recipient: "xrn:sdgkjhs2345u79ghisodg", tradable_amount: "10", retired_amount: "2", retirement_location: "YY-ZZ 12345"}]'
-              Note: "tradable_amount" and "retired_amount" default to 0.
-              Note: "retirement_location" is only required when "retired_amount" is positive.`,
-		Args: cobra.ExactArgs(6),
+  msg-create-batch: A JSON object representing MsgCreateBatch. This could be
+                    passed from a JSON file like:
+		    regen tx create-batch "$(< batch.json)"
```

Comment:
why not just using `io/ioutil` instead (see for instance in x/group `parseMembers` function) and you would just need the following, which is more straightforward from a user standpoint and aligned with how it's done in the sdk tx commands: 
```suggestion
		    regen tx create-batch batch.json"
```